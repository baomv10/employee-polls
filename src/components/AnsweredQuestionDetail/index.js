import { useMemo } from "react";

const AnsweredQuestionDetail = ({ question }) => {
  const optionPercentage = useMemo(() => {
    if (!question) return 0;
    const optionOne = question.optionOne.votes.length;
    const optionTwo = question.optionTwo.votes.length;
    const total = optionOne + optionTwo;
    const result = {
      optionOne: ((optionOne / total) * 100).toFixed(0),
      optionTwo: ((optionTwo / total) * 100).toFixed(0),
    };
    return result;
  }, [question]);

  return (
    <>
      <div className="col-6 text-center">
        <div className="border rounded p-1">
          <p> {question?.optionOne?.text}</p>
        </div>
        <p>{question?.optionOne.votes.length}</p>
        <p>{optionPercentage.optionOne}%</p>
      </div>
      <div className="col-6 text-center">
        <div className="border rounded p-1">
          <p> {question?.optionTwo?.text}</p>
        </div>
        <p>{question?.optionTwo.votes.length}</p>
        <p>{optionPercentage.optionTwo}%</p>
      </div>
    </>
  );
};

export default AnsweredQuestionDetail;
