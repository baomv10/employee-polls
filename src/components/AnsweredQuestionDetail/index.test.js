import * as React from "react";
import { render } from "@testing-library/react";
import AnsweredQuestionDetail from ".";

describe("AnsweredQuestionDetail", () => {
  it("matches the snapshot when question is passed", () => {
    var component = render(
      <AnsweredQuestionDetail
        question={{
          id: "8xf0y6ziyjabvozdd253nd",
          author: "sarahedo",
          timestamp: 1467166872634,
          optionOne: {
            votes: ["sarahedo"],
            text: "Build our new application with Javascript",
          },
          optionTwo: {
            votes: [],
            text: "Build our new application with Typescript",
          },
        }}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it("matches the snapshot when no question is passed", () => {
    var component = render(<AnsweredQuestionDetail />);
    expect(component).toMatchSnapshot();
  });
});
