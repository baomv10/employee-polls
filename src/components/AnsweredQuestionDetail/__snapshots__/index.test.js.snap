// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnsweredQuestionDetail matches the snapshot when a name is passed 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="col-6 text-center"
      >
        <div
          class="border rounded p-1"
        >
          <p>
             
            Build our new application with Javascript
          </p>
        </div>
        <p>
          1
        </p>
        <p>
          100
          %
        </p>
      </div>
      <div
        class="col-6 text-center"
      >
        <div
          class="border rounded p-1"
        >
          <p>
             
            Build our new application with Typescript
          </p>
        </div>
        <p>
          0
        </p>
        <p>
          0
          %
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="col-6 text-center"
    >
      <div
        class="border rounded p-1"
      >
        <p>
           
          Build our new application with Javascript
        </p>
      </div>
      <p>
        1
      </p>
      <p>
        100
        %
      </p>
    </div>
    <div
      class="col-6 text-center"
    >
      <div
        class="border rounded p-1"
      >
        <p>
           
          Build our new application with Typescript
        </p>
      </div>
      <p>
        0
      </p>
      <p>
        0
        %
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`AnsweredQuestionDetail matches the snapshot when no name is passed 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="col-6 text-center"
      >
        <div
          class="border rounded p-1"
        >
          <p>
             
          </p>
        </div>
        <p />
        <p>
          %
        </p>
      </div>
      <div
        class="col-6 text-center"
      >
        <div
          class="border rounded p-1"
        >
          <p>
             
          </p>
        </div>
        <p />
        <p>
          %
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="col-6 text-center"
    >
      <div
        class="border rounded p-1"
      >
        <p>
           
        </p>
      </div>
      <p />
      <p>
        %
      </p>
    </div>
    <div
      class="col-6 text-center"
    >
      <div
        class="border rounded p-1"
      >
        <p>
           
        </p>
      </div>
      <p />
      <p>
        %
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`AnsweredQuestionDetail matches the snapshot when no question is passed 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="col-6 text-center"
      >
        <div
          class="border rounded p-1"
        >
          <p>
             
          </p>
        </div>
        <p />
        <p>
          %
        </p>
      </div>
      <div
        class="col-6 text-center"
      >
        <div
          class="border rounded p-1"
        >
          <p>
             
          </p>
        </div>
        <p />
        <p>
          %
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="col-6 text-center"
    >
      <div
        class="border rounded p-1"
      >
        <p>
           
        </p>
      </div>
      <p />
      <p>
        %
      </p>
    </div>
    <div
      class="col-6 text-center"
    >
      <div
        class="border rounded p-1"
      >
        <p>
           
        </p>
      </div>
      <p />
      <p>
        %
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`AnsweredQuestionDetail matches the snapshot when question is passed 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="col-6 text-center"
      >
        <div
          class="border rounded p-1"
        >
          <p>
             
            Build our new application with Javascript
          </p>
        </div>
        <p>
          1
        </p>
        <p>
          100
          %
        </p>
      </div>
      <div
        class="col-6 text-center"
      >
        <div
          class="border rounded p-1"
        >
          <p>
             
            Build our new application with Typescript
          </p>
        </div>
        <p>
          0
        </p>
        <p>
          0
          %
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="col-6 text-center"
    >
      <div
        class="border rounded p-1"
      >
        <p>
           
          Build our new application with Javascript
        </p>
      </div>
      <p>
        1
      </p>
      <p>
        100
        %
      </p>
    </div>
    <div
      class="col-6 text-center"
    >
      <div
        class="border rounded p-1"
      >
        <p>
           
          Build our new application with Typescript
        </p>
      </div>
      <p>
        0
      </p>
      <p>
        0
        %
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
